mars/optimization/logical/tileable/column_pruning/self_column_selector.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Set, Any, Callable
+
+from .utils import get_cols_exclude_index
+from .....core import TileableData
+from .....dataframe.core import BaseDataFrameData, BaseSeriesData
+from .....dataframe.groupby.aggregation import DataFrameGroupByAgg
+from .....dataframe.indexing.getitem import DataFrameIndex
+from .....dataframe.indexing.setitem import DataFrameSetitem
+from .....dataframe.merge import DataFrameMerge
+from .....typing import OperandType
+
+
+class SelfColumnSelector:
+
+    _OP_TO_SELECT_FUNCTION = {}
+
+    @classmethod
+    def register(
+        cls,
+        op_cls: OperandType,
+        func: Callable[[TileableData], Set[Any]],
+    ) -> None:
+        if op_cls not in cls._OP_TO_SELECT_FUNCTION:
+            cls._OP_TO_SELECT_FUNCTION[op_cls] = func
+        else:
+            raise ValueError(f"key {op_cls} exists.")
```

Comment:
IMO, override is better than raising errors if key exists, it can be patched if something is wrong.